src/Persistence/PaYaml/Serialization/PaYamlSerializer.cs

```diff
@@ -146,21 +150,25 @@ public static bool CheckIsSequence(string yaml)
     /// <summary>
     /// checks the input source is a sequence of YAML items.
     /// </summary>
+    /// <remarks>
+    /// This method does not perform validation on the input fragment.
+    /// </remarks>
     /// <param name="reader"></param>
     /// <returns></returns>
-    /// <exception cref="ArgumentNullException"></exception>
+    /// <exception cref="ArgumentNullException">thrown when <paramref name="reader"/> is null.</exception>
     public static bool CheckIsSequence(StringReader reader)
     {
         ArgumentNullException.ThrowIfNull(reader);
 
-        var builder = new DeserializerBuilder()
-            .IgnoreUnmatchedProperties()
-            .WithTypeConverter(new YamlSequenceTesterConverter());
-        var deserializer = builder.Build();
         try
```

Comment:
I don't think we should need to put the try/catch here as the code uses the TryConsume methods only, which shouldn't throw